### PR TITLE
feat: fork `react-dropzone`

### PR DESCRIPTION
**Pull request by juliusmarminge**
This is a straight-up fork of react-dropzone (incl. rewrite in TypeScript and dropped usage of PropTypes & defaultProps) with better tooling and ESM support. The original package is unmaintained for almost a year now, and doesn't seem to be interested in merging existing PRs

There's some `@ts-expect-error`'s in there as I got tired of fixing everything 😅

- [ ] Port tests?
- [ ] `<Dropzone />`?
- [ ] Improve package code?

### CODE REVIEW

**Comment by Mr0Bread** [comment]
That's how packages become 'zero dependencies' type ;DDD

**Review by t3dotgg — APPROVED** [review]
lgtm

**Review by markflorkowski — APPROVED** [review]
Seems reasonable to me 👍

In the longer term, should we potentially consider integrating the dropzone code directly into our component rather than having a separate package? 

cc @t3dotgg

**Comment by juliusmarminge** [comment]
> In the longer term, should we potentially consider integrating the dropzone code directly into our component rather than having a separate package?

I didn't do this initially cause I thought everyone didn't need that code, but then i realized if you're generating components, you will get the dropzone code imported whether you like it or not... So yea, this could just as well be in the main package if we don't care about external consumers might wanna use it without uploadthing?

**Comment by markflorkowski** [comment]
> So yea, this could just as well be in the main package if we don't care about external consumers might wanna use it without uploadthing?

Yeah, I think it would probably better this way, as we could cut out anything we don't actually need and not worry about maintaining *another* package for external consumption.

File: docs/src/pages/faq.mdx [hunk 1]
```diff
@@ -27,8 +27,3 @@ their `middleware.ts` is configured incorrectly.
 Make sure that `/api/uploadthing` is NOT blocked through your middleware. It
 serves as both a validation endpoint from client AND a webhook for the
 `onUploadComplete` calls.
-
-### I'm getting the error `Named export 'useDropzone' not found`
```
Comment (markflorkowski) [inline docs/src/pages/faq.mdx:31]: This is a *really* nice benefit.

**Review by markflorkowski — APPROVED** [review]
Excited to no longer have this as a dep. We should probably add a comment at the top of the `use-dropzone/index.tsx` making note that it is a fork and where it came from.

**Review by markflorkowski — APPROVED** [review]

**Pull request merged**